File: src/reset.py
import os
import time
import textwrap
import scripts

def resetter():
    save = scripts.savePull()
    var9 = save[8]
    var10 = save[9]
    if var9 == 0:
        print('\nDoing this will reset everything. Are you sure?')
        print('\nRESET [1]')
        print('BACK [2]')
    if var9 == 1:
        print('\nRESET [1]')
    while True:
        try:
            choose = int(input('\nACTION >> '))
            if choose == 1:
                if var9 == 1:
                    scripts.savePull()
                    print('\nReset failed.\n')
                    time.sleep(1)
                    if var10 == 0:
                        line_ext = 9
                        state_ext = 1
                        scripts.saveWriter()
                    quit()
                if var9 == 0:
                    normalReset()
                    return
            if choose == 2:
                if var9 == 1:
                    quit()
                if var9 == 0:
                    return
            if choose == 420:
                print('')
                scripts.easterEgg()
                resetter()
        except ValueError:
            scripts.inpErrorHandler()
            resetter()

def normalReset():
    line_ext = 0
    state_ext = 0
    scripts.saveWriter(line_ext, state_ext)
    line_ext = 1
    state_ext = 0
    scripts.saveWriter(line_ext, state_ext)
    line_ext = 2
    state_ext = 0
    scripts.saveWriter(line_ext, state_ext)
    line_ext = 3
    state_ext = 0
    scripts.saveWriter(line_ext, state_ext)
    line_ext = 4
    state_ext = 1
    scripts.saveWriter(line_ext, state_ext)
    line_ext = 5
    state_ext = 0
    scripts.saveWriter(line_ext, state_ext)
    line_ext = 7
    state_ext = 0
    scripts.saveWriter(line_ext, state_ext)
    line_ext = 13
    state_ext = 0
    scripts.saveWriter(line_ext, state_ext)
    line_ext = 14
    state_ext = 0
    scripts.saveWriter(line_ext, state_ext)
    line_ext = 15
    state_ext = 0
    scripts.saveWriter(line_ext, state_ext)
    print('\nReset.')
    time.sleep(0.5)

File: src/reset.cpp
#include <iostream>
#include <fstream>
#include <unistd.h>
#include <time.h>
#include "backend.hpp"
#include "director.hpp"
#include "visuals.hpp"

using namespace std;

void std_reset(){
	ofstream x("data.txt");
	x<<"0\n0\n0\n0\n1\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0\n0";
	x.close();
}

void resetter(){
	screen_clear();
	int * x = save_reader();
	int v9 = x[8];
	int i;
	if ( v9 == 0 ){
		cout << "\n" << "Doing this will reset everything. Are you sure?" << endl;
		cout << "\n" << "RESET [1]" << endl;
		cout << "BACK [2]" << endl;
	}
	if ( v9 == 1 ){
		cout << "\n" << "RESET [1]" << endl;
	}
	cout << "\nACTION >> ";
	cin >> i;
	if ( i ==  1 ){
		if ( v9 == 0 ){
			std_reset();
			cout << "\n" << "Reset." << endl;
			sleep(1);
			direct();
		}
		else{}
	}
}


	
